### PR TITLE
use latest libcfbuildpack

### PR DESCRIPTION
**Pull request by elbandito**
This is a short term PR to unblock the issues we're having with the out-of-date 3rd party libraries.  For now, the plan is to use the latest version of the `libcfbuildpack`, and to ignore all dependency enforcements on other buildpacks e.g. npm/node.   The dependency issues will be addressed in later PRs to take advantage of the new `Provides` and `Requires` features of the CNB spec.

### CODE REVIEW

**Comment by jabrown85** [comment]
What is the blocking issue? I think I understand what you are saying but I haven't seen the details.

**Comment by elbandito** [comment]
@jabrown85 The biggest blocking issue is how CNB switched to the `Provides` and `Requires` schema.  Currently, we're just adding dependencies into a Map via the old `libhkbuildpack` interface, and that's all broken now.  When we update to use this new schema, we may not even need to programmatically check that a dependency is being met, the Lifecycle will enforce this for us by specifying:

```
[[requires]]
name = "<dependency name>"
version = "<dependency version>"
```

@hone is there anything else to add?

File: node/buildpack_test.go [hunk 1]
```diff
@@ -44,161 +38,164 @@ func TestName(t *testing.T) {
...
-			f.AddBuildPlan(Dependency, buildplan.Dependency{})
-			f.AddDependency(Dependency, ".")
-
-			err := b.Build(f.Build)
-
-			g.Expect(err).To(BeNil())
-		})
-	}, spec.Report(report.Terminal{}))
-}
+//func TestDetect(t *testing.T) {
```
Comment (hone) [inline node/buildpack_test.go]: should we just have a simple test here vs commenting it all out?

**Comment by hone** [comment]
This looks good to me as a first pass. I left a comment about the tests as a suggestion, but maybe not worth the time. We can do proper build plan stuff in another PR.

**Pull request merged**